service/file_controller_create_volume.go

```diff
@@ -186,3 +206,68 @@ func queryShareData(conn RestEndpoint, fsName string) (map[string]string, error)
 	}
 	return nfsShareMetaData, err
 }
+
+func queryFileInterface(conn RestEndpoint, nasId string) (*list.List, error){
+	var getFileInterfaceUrl string = "/api/types/fileInterface/instances?fields=id,nasServer"
+    encodedQuery := EncodeUrl(getFileInterfaceUrl)
+    logrus.Debug("Encoded: ", encodedQuery)
+    status, resp := conn.get(encodedQuery)
+    logrus.Debug("Status: ", status)
+    logrus.Debug("Query file interface response: ", resp)
+    jsonParsed, _ := gabs.ParseJSON([]byte(resp))
+    entryCount := jsonParsed.Path("entryCount").Data().(float64)
+    fileInterfaces := list.New();
+
+
+    var err error = nil
+    if int(entryCount) == 0 {
+    	logrus.Info("Info: the num of file interface entries is 0")
+    	//err = errors.New(fmt.Sprintf("Error: no file interface created for nas server:%s ", nasId))
+	}else{
+		children, _ := jsonParsed.S("entries").Children()
+		for _, child := range children{
+			if child.Path("content").Path("nasServer").Data().(string) == nasId {
+				fileInterfaceData := make(map[string]string)
+				fileInterfaceData["id"] = child.Path("content").Path("id").Data().(string)
+				fileInterfaces.PushBack(fileInterfaceData);
+				logrus.Info(fileInterfaceData)
+			}
+		}
+		logrus.Info(fileInterfaces)
+	}
+    return fileInterfaces, err
+}
+
+func createFileInterface(conn RestEndpoint, fileIPInput fileInterfaceInput)(map[string]string, error) {
+	var err error = nil
+	fileInterfaceData := make(map[string]string)
+
+	fileInterfaces, _ := queryFileInterface(conn, fileIPInput.nasServer)
+	if fileInterfaces.Len() == 0 {
+		logrus.Info("Info: file interface is not created for nas server %s", fileIPInput.nasServer)
+		// start to create interface
+		var createFileInterfaceUrl string = "/api/types/fileInterface/instances"
```

Comment:
I suggested to use async mode for file interface creation by adding timeout=0.
In sync mode, If the create request last a long time waiting and there is a network issue happened, you may lose the created file interface id.
If you use async mode, you can get a job id quickly. Then you can use this job id to get the creation result any time.

Reply:
Thanks for comments, updated with async call in new commit.